apps/gallery/js/ImageEditor.js

```diff
@@ -371,10 +400,14 @@ function setEditTool(editTool) {
       showEditToolView();
       $('edit-crop-options').classList.remove('hidden');
       $('edit-title').setAttribute('data-l10n-id', 'crop');
-      // Show crop overlay on entering crop edit tool
-      imageEditor.edit(function() {
-        imageEditor.showCropOverlay();
-      });
+        // Initialize the source rectangle to full size of the original image
+        imageEditor.resetCropRegion();
+        // Show screen with crop overlay on entering crop edit tool
+        imageEditor.requestNewPreview(function() {
+          // Use saved crop region in editSettings to show crop overlay
+          imageEditor.showCropOverlay(
+            imageEditor.getCropOverlayRegion(savedEditSettings.crop));
```

Comment:
At this point editSettings and savedEditSettings are still the same, right? I'd think that it is better to use editSettings here, since you're not trying to go back and restore anything.